strategies/affected/src/main/java/org/arquillian/smart/testing/strategies/affected/ClassFileIndex.java

```diff
@@ -114,66 +112,59 @@ private void replaceVertex(JavaElement newClass) {
 
     private void updateJavaElementWithImportReferences(JavaElement javaElementParentClass,
         String[] imports) {
+
         for (String importz : imports) {
+
+            if (addImport(javaElementParentClass, importz)) {
+                if (filter.shouldBeIncluded(importz)) {
+                    JavaClass javaClass = builder.getClass(importz);
+                    if (javaClass != null) {
+                        updateJavaElementWithImportReferences(javaElementParentClass, javaClass.getImports());
+                    }
+                }
+            }
+        }
+    }
+
+    private boolean addImport(JavaElement javaElementParentClass, String importz) {
+
+        if (!importz.startsWith("java")) {
+
             JavaElement importClass = new JavaElement(importz);
             if (!importClass.equals(javaElementParentClass)) {
                 if (!graph.containsVertex(importClass)) {
                     graph.addVertex(importClass);
                 }
+
+                // This condition is required because we are flattering imports in graph
+                // This operation is fast since it is a containsKey call in a Map instance
+                if (!graph.containsEdge(javaElementParentClass, importClass)) {
+                    graph.addEdge(javaElementParentClass, importClass);
+                    return true;
+                }
                 graph.addEdge(javaElementParentClass, importClass);
```

Comment:
What is this code really doing?

`boolean addImport(...)` - I would assume `boolean` value here indicates if adding connection was performered or not. However it's not what this code really does. Line `142` and `145` do the same, but in case of condition defined in `141` is not satisfied we add an `edge` (or overwrite it) but return `false`? Then maybe we could simple change this to:

```java
boolean edgeAlreadyDefined = graph.containsEdge(javaElementParentClass, importClass);
graph.addEdge(javaElementParentClass, importClass);
return !edgeAlreadyDefined;
```

But then still there is something blurry in this whole concept.